src/main/java/io/wispforest/owo/serialization/impl/CooptCodeck.java

```diff
@@ -0,0 +1,109 @@
+package io.wispforest.owo.serialization.impl;
+
+import com.mojang.datafixers.util.Pair;
+import com.mojang.serialization.Codec;
+import com.mojang.serialization.DataResult;
+import com.mojang.serialization.DynamicOps;
+import com.mojang.serialization.JsonOps;
+import io.wispforest.owo.serialization.Codeck;
+import io.wispforest.owo.serialization.Deserializer;
+import io.wispforest.owo.serialization.Serializer;
+import io.wispforest.owo.serialization.impl.json.JsonDeserializer;
+import io.wispforest.owo.serialization.impl.json.JsonSerializer;
+import io.wispforest.owo.serialization.impl.nbt.NbtDeserializer;
+import io.wispforest.owo.serialization.impl.nbt.NbtSerializer;
+import net.minecraft.nbt.NbtOps;
+import net.minecraft.util.dynamic.ForwardingDynamicOps;
+import org.jetbrains.annotations.Nullable;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.function.Function;
+import java.util.function.Supplier;
+
+public class CooptCodeck<T> implements Codeck<T>, Codec<T> {
+
+    public static Map<DynamicOps<?>, Pair<Supplier<Serializer<Object>>, Function<Object, Deserializer<Object>>>> MAP = new HashMap<>();
+
+    static {
+        register(JsonOps.INSTANCE, JsonSerializer::new, JsonDeserializer::new);
+        register(NbtOps.INSTANCE, NbtSerializer::new, NbtDeserializer::new);
+    }
+
+    private final Codeck<T> codeck;
+
+    public CooptCodeck(Codeck<T> codeck){
+        this.codeck = codeck;
+    }
+
+    private static <V> void register(DynamicOps<?> ops, Supplier<Serializer<V>> supplier, Function<V, Deserializer<V>> function){
+        MAP.put(ops, new Pair<>(() -> (Serializer<Object>) supplier.get(), o -> (Deserializer<Object>) function.apply((V) o)));
+    }
+
+    @Nullable
+    private static <T1> Pair<Supplier<Serializer<Object>>, Function<Object, Deserializer<Object>>> convertToFormat(DynamicOps<T1> ops){
+        if(ops instanceof ForwardingDynamicOps<T1> forwardDynOps) ops = forwardDynOps.delegate;
+
+        return MAP.containsKey(ops) ? MAP.get(ops) : new Pair<>(null, null);
+    }
+
+    @Override
+    public <T1> DataResult<Pair<T, T1>> decode(DynamicOps<T1> ops, T1 input) {
+        var deserializer = (Deserializer<T1>) convertToFormat(ops).getSecond().apply(input);
+
+        boolean error = true;
+        Exception exception = null;
+
+        T value = null;
+
+        if(deserializer != null) {
+            try {
+                value = decode(deserializer);
+
+                error = false;
+            } catch (Exception e) {
+                exception = e;
+            }
+        }
+
+        return error
+                ? DataResult.error(exception != null ? exception::getMessage : () -> "The corresponding Format for the given DynamicOps could not be located!")
```

Comment:
This shouldn't really be a problem, as if you get a funky DynamicOps you can always just NbtOps or JsonOps and then convert everything to the needed dynamic ops

Reply:
True but I rather have this error instead to properly map such due to how conversion between formats can cause a lose of information potentially